### ISSUE TITLE
Marriage Report Filter Bug

### ISSUE THREAD

**Issue by miqrogroove** [fisharebest/webtrees#4919]
I'm running a marriage report with the following criteria ...

Place: Michigan
Start: 1/1/1944
End: 12/31/1944

There are 9 families listed in the report.  All of the results are correct except for the 2nd-to-last entry.  It's showing a marriage that took place in Indiana during 1936.

Running webtrees v2.1.18 on PHP v8.3.0.

**Comment by miqrogroove** [comment]
So here's what I'm thinking ...

https://github.com/fisharebest/webtrees/blob/e96e110d1e1c5e1d8eac6806f311b102a73c1907/resources/xml/reports/marriage_report.xml#L60

This filter is applied at the Family level, rather than the Marriage level.

In my test case, the Family contains 2 different marriages, as follows:

```
0 @F1@ FAM
1 MARR
2 DATE 1936
2 PLAC Indiana, USA
1 MARR
2 DATE 1945
2 PLAC Michigan, USA
```

Now, while neither of those marriages can match my search criteria, in fact the Family does match.

The place `filter1` matches the 2nd marriage.  The start date `filter2` matches the 2nd marriage.  The end date `filter3` matches the 1st marriage.

The result is a false positive.

**Comment by arbor95** [comment]
I think the possibilty to enter more than one date for the marriage ( without using a specifying type), that is establishing the family, should not be possible (and is imho a gedcom design bug), although family in gedcom has an other meaning than we normaly use.
If I do not know the correct date, I would explain that within a note.

**Comment by miqrogroove** [comment]
@arbor95 You've got the wrong topic.  This issue is about families with multiple marriages.  To be clear, I am only talking about "a family" where the couple was married more than once.

**Comment by arbor95** [comment]
You have to create a new family in that rare cases

**Comment by miqrogroove** [comment]
@arbor95 Clearly we are not talking about the same thing.  Feel free to start a separate thread in the forum.

**Comment by arbor95** [comment]
No, if your gedcom - input is wrong, you can not expect correct results.
